src/SuperVaultFactory.sol

```diff
@@ -99,46 +70,51 @@ contract SuperVaultFactory is ISuperVaultFactory, AccessControl {
         }
 
         bytes32 salt = keccak256(abi.encodePacked(asset_, name_, superformIds_, startingWeights_, "SuperVault"));
-
-        address superVault = address(
+        superVault = address(
             new SuperVault{ salt: salt }(
-                address(superRegistry),
-                asset_,
-                strategist_,
-                vaultManager_,
-                name_,
-                depositLimit_,
-                superformIds_,
-                startingWeights_
+                superRegistry, asset_, strategist_, vaultManager_, name_, depositLimit_, superformIds_, startingWeights_
             )
         );
 
+        if (formImplementationId4626_ == 0 || formImplementationId5115_ == 0) {
+            revert ZERO_FORM_IMPLEMENTATION_ID();
+        }
+
+        _setValidFormImplementationId(superVault, formImplementationId4626_);
+        _setValidFormImplementationId(superVault, formImplementationId5115_);
+
+        /// @dev set performance fee to 0
+        (bool success,) = address(superVault).call(abi.encodeCall(ITokenizedStrategy.setPerformanceFee, (0)));
+        if (!success) {
+            revert FAILED_TO_SET_PERFORMANCE_FEE();
+        }
+
         /// @dev set pending management to deployer
         /// @dev deployer will have to accept management in SuperVault
-        (bool success,) = address(superVault).call(
-            abi.encodeWithSelector(ITokenizedStrategy.setPendingManagement.selector, msg.sender)
-        );
+        (success,) = address(superVault).call(abi.encodeCall(ITokenizedStrategy.setPendingManagement, (msg.sender)));
         if (!success) {
             revert FAILED_TO_SET_PENDING_MANAGEMENT();
         }
 
         superVaults.push(superVault);
-        registeredSuperVaults[superVault] = true;
 
-        return superVault;
+        emit SuperVaultCreated(superVault);
     }
 
     //////////////////////////////////////////////////////////////
     //                  EXTERNAL VIEW FUNCTIONS                 //
     //////////////////////////////////////////////////////////////
 
     /// @inheritdoc ISuperVaultFactory
-    function isSuperVault(address superVault_) external view returns (bool) {
-        return registeredSuperVaults[superVault_];
+    function getSuperVaults() external view override returns (address[] memory) {
```

Comment:
With a huge number of `superVaults` this could out of gas.

possible solution: make `superVaults[]` public and add a function to retrieve `superVaults.length`.

Reply:
Oh good point thanks

Reply:
Fixed in 289b21a